Remove code related to Puppet version 4

wbclark opened the pull request:
Puppetserver 4 was only present on Satellite 6.3. At present, the oldest
supported Satellite release is Satellite 6.5. So code related to
Puppetserver 4 should be removed.

ekohl commented on library/parse_backup_metadata.py:
```diff
@@ -56,7 +47,6 @@ def parse_backup_metadata(params):
     satellite_version = get_rpm_version(rpms, "^satellite-[\d+].*")
     puppet_agent_version = get_rpm_version(rpms, "^puppet-agent-[\d+].*", 2, 1)
     puppet_rpm_version = get_rpm_version(rpms, "^puppet-[\d+].*", 1, 1)
```
These variables are now unused

ekohl commented on roles/satellite-clone/tasks/main.yml:
```diff
@@ -166,15 +160,6 @@
       permissions. Avoid using /root.
   when: clone_foreman_dump_exists and clone_candlepin_dump_exists and (clone_no_foreman_dump_access or clone_no_candlepin_dump_access)
 
-# If there is no puppet user, the extracted config files will not have the correct puppet
-# user/group permissions with puppet 4. Installing puppetserver before extracting config_files
-# fixes this issue.
-- name: Install puppetserver to create puppet user
-  yum:
-    name: puppetserver
-    state: latest
-  when: puppet_version|int == 4
```
Isn't this still true for all Puppetserver versions today?

wbclark replied:
If that's true then there is a bug here, because the above task only runs `when: puppet_version|int == 4`

I wonder what is the historical reason this was needed (can check blame/logs to have a look), as I think the installer should ensure we have correct user/group/permissions as long as we reach that point

wbclark replied:
Git forensics reveals the issue was: https://github.com/RedHatSatellite/satellite-clone/issues/328

The issue report states what was the fix, but does not state what was the original problem

It was resolved in https://github.com/RedHatSatellite/satellite-clone/commit/b0de5e6983b5ba58499cba63136b034301b96960 with a comment at https://github.com/RedHatSatellite/satellite-clone/pull/333#issuecomment-397024596 mentioning permissions issues with puppet, but not further details

In my opinion it's safe to remove this, because this code already doesn't run on any of the currently supported Satellite/Puppet versions

evgeni replied:
Given nobody complained for a while… ;)

ekohl replied:
I can imagine that the puppet user should be present for the backup restore to succeed, but it could very well be that the first installer run rectifies all the permissions.

ekohl reviewed (changes requested):
In description I'd just use `Puppet version 4`.

wbclark commented:
Thanks @ekohl . I've updated to use "Puppet" over "puppetserver" in the commit message, and removed the additional unused variables.

Investigating why the puppetserver pre-install was ever needed to begin with and whether it should in fact run with other Satellite/Puppet versions

ekohl reviewed (approved):
I don't see how the old code could have worked once we upgraded to Puppet 5 so let's drop it.

evgeni commented:
Agreed
